### PR TITLE
fixing download popup for virtual firealarm

### PR DESCRIPTION
**Pull request by rasika90**

### CODE REVIEW

File: features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/private/conf/device-type.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "deviceType": {
     "displayName": "Virtual Firealarm",
-    "category" : "iot"
+    "category": "iot",
+    "downloadAgentUri": "https://localhost:9443/virtual_firealarm/VirtualFireAlarmDeviceManager/manager/device/virtual_firealarm/download",
```
Comment (prabathabey) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/private/conf/device-type.json]: Shouldn't all these URLs be parameterized?

Comment (rasika90) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/private/conf/device-type.json]: No. This is an UUF unit component associated with a single deviceType which is "VirtualFirealarm". Any DeviceType writer can have their own download link url.

Comment (milanperera) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/private/conf/device-type.json]: But, what happens if we change the port offset from carbon.xml?
Then the 9443 port wont work.

File: features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs [hunk 2]
```diff
@@ -288,7 +268,7 @@
         function toggleEnrollment(){
             $(".modalpopup-content").html($("#qr-code-modal").html());
             //generateQRCode(".modalpopup-content .qr-code");
-            showPopup();
+            //showPopup();
```
Comment (milanperera) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs]: Why do we keep commented code line here?

File: features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/private/conf/device-type.json (hunk 1 above)
Comment (rasika90) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/private/conf/device-type.json]: Agreed. but this config doesn't exist now. This port offset issue is addressed with handlebar parameter `{{@app.context}}` which returns `http://<server ip>:<server port>/<app context>`.

File: features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs (hunk 2 above)
Comment (rasika90) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs]: This is already fixed in [#826d427](https://github.com/wso2/carbon-device-mgt-plugins/commit/826d427143b29b25e2d198cedcb3e778cdf6a0e0).

File: features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs [hunk 3]
```diff
@@ -38,60 +38,37 @@
     <div id="download-device-modal-content" class="hide">
         <div class="modal-content">
             <div class="row">
-                <div class="col-md-7 col-centered center-container">
-                    <h3>Name your device and download the agent from following link.</h3>
+                <div class="col-md-7  center-container">
+                    <h3>Name your device and download the agent from following link.dsfsdfdsf</h3>
```
Comment (DilanUA) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs]: what does this "dsfsdfdsf" stand for?

Comment (rasika90) [inline features/device-mgt-iot-virtualfirealarm-feature/org.wso2.carbon.device.mgt.iot.virtualfirealarm.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.virtual_firealarm.type-view/type-view.hbs]: Already fixed with [#6e64566](https://github.com/wso2/carbon-device-mgt-plugins/commit/6e645667b11a97440b4dc65697ad5f80d273a124).

**Pull request merged**